src/D2L.Security.OAuth2/Keys/Default/LocalPublicKeyProvider.cs

```diff
@@ -20,6 +21,27 @@ IInMemoryPublicKeyCache cache
 			m_cache = cache ?? throw new ArgumentNullException( nameof( cache ) );
 		}
 
+		async Task IPublicKeyProvider.PrefetchAsync() {
+			IEnumerable<JsonWebKey> jwks = await m_publicKeyDataProvider
+				.GetAllAsync()
+				.ConfigureAwait( false );
+
+			foreach( JsonWebKey jwk in jwks ) {
+				if( m_cache.Get( PUBLIC_KEY_SOURCE, jwk.Id ) is not null ) {
+					continue;
+				}
+
+				D2LSecurityToken token;
+				try {
+					token = jwk.ToSecurityToken();
+				} catch {
+					continue;
+				}
+
+				m_cache.Set( PUBLIC_KEY_SOURCE, token );
+			}
+		}

```

Comment:
🤔 this impl is probably unused? OK tho

Reply:
I guess it could be used tho. It wouldn't be a big win currently but I wouldn't object.

Reply:
Yeah, I noted that in my folded/Outdated comment above

> NOTE: This class is only used by the auth service for the validate route. Not planning on actually leveraging it at the moment. I didn't add tests for this (but am willing to).

(I did add the tests just now).

Implemented it because it was being added to the interface. Don't super expect us to use it (though auth service could do so on task boot, :shrug:).